### PR TITLE
Improve account failure state handling and logging

### PR DESCRIPTION
**Pull request by reachjainrahul**
Store account failure state in account config struct inside plugin so that CPA/CES status can be updated with correct info. Also changed logging of some error logs from info to error.

### CODE REVIEW

**Comment by reachjainrahul** [comment]
nephe-test-e2e-agentless-upgrade

File: pkg/cloudprovider/plugins/internal/accounts_common.go [hunk 1]
```diff
@@ -137,9 +142,13 @@ func (c *cloudCommon) updateCloudAccountConfig(client client.Client, credentials
 		return err
 	}
 
-	if err := currentConfig.serviceConfig.UpdateServiceConfig(serviceConfig); err != nil {
-		cloudAccountConfigState = false
-		return err
+	if currentConfig.serviceConfig == nil {
+		currentConfig.serviceConfig = serviceConfig
+	} else {
```
Comment (shenmo3) [inline pkg/cloudprovider/plugins/internal/accounts_common.go:147]: Consider move the update outside of the else block and just return nil after L146 in the if block.

File: pkg/cloudprovider/plugins/internal/accounts_common.go [hunk 2]
```diff
@@ -78,40 +78,45 @@ func (c *cloudCommon) newCloudAccountConfig(client client.Client, namespacedName
...
-		// Update the cloud account config state for each update.
-		currentConfig.state = cloudAccountConfigState
-	}()
 
 	credentialsValidatorFunc, cloudServicesCreateFunc, credentialsComparatorFunc, err := c.getFunctionPointers()
 	if err != nil {
 		cloudAccountConfigState = false
 		return err
 	}
+	defer func() {
```
Comment (shenmo3) [inline pkg/cloudprovider/plugins/internal/accounts_common.go:113]: we don't need to update state if error happen on getting func pointers?

File: pkg/cloudprovider/plugins/internal/accounts_common.go (hunk 1 above)
Comment (reachjainrahul) [reply pkg/cloudprovider/plugins/internal/accounts_common.go:147]: We want to continue if no error occurs. For example, if someone adds some code at line 153, that won't be executed.

**Review by shenmo3 — APPROVED** [review]

**Pull request merged**